javascript/javascript2/week3/fastest-presser.js

```diff
@@ -0,0 +1,72 @@
+/*
+When the button is clicked, get the value of the input.
+This value will be the amount of time the game should run.
+*/
+
+const gameButton = document.querySelector("#start"); //start game button
+const input = document.querySelector("input");
+let sCount = 0;
+let lCount = 0;
+
+const resultTag = document.getElementById("game-result"); //game result
+resultTag.innerHTML = "";
+const gameResult = () => {
+  if (sCount == 0 && lCount == 0) {
+    resultTag.innerHTML = "No action!!!";
+  }
+  if (sCount == lCount && sCount > 0 && lCount > 0) {
+    resultTag.innerHTML = "Game Draw!!!";
+  }
+  if (sCount > lCount) {
+    resultTag.innerHTML = "S is the winer !!!";
+
+    const confettiSettings = { target: "s-canvas", size: 2 };
+    const confetti = new ConfettiGenerator(confettiSettings);
+    confetti.render();
+  }
+  if (lCount > sCount) {
+    resultTag.innerHTML = "L is the winner";
+    const confettiSettings = { target: "l-canvas", size: 2 };
+    const confetti = new ConfettiGenerator(confettiSettings);
+    confetti.render();
+  }
+};
+
+gameButton.addEventListener("click", gameStart);
+function gameStart() {
+  let gameTime = input.value;
+  if (gameTime == "") {
+    alert("please enter time");
+    document.removeEventListener("keydown", keyDetect);
+  }
+
+  if (gameTime > 0) {
+    setTimeout(() => {
+      document.removeEventListener("keydown", keyDetect);
+      alert(" Game End!!!");
+      gameResult();
+    }, gameTime * 1000);
+
+    document.addEventListener("keydown", keyDetect); //detect the keys
+    function keyDetect(event) {
+      let userKey = event.key;
+      if (userKey == "s") {
+        sCount++;
+        document.getElementById("S-press").innerHTML = sCount;
+      }
+      if (userKey == "l") {
+        lCount++;
+        document.getElementById("L-press").innerHTML = lCount;
+      }
+    }
+  }
+}
+document.getElementById("restart").addEventListener("click", function () {
+  document.getElementById("input").value = "";
+  document.getElementById("S-press").innerHTML = "";
+  document.getElementById("L-press").innerHTML = "";
+  resultTag.innerHTML = "";
+  sCount = 0;
+  lCount = 0;
+  confetti.clear();
```

Comment:
This `confitti.clear()` it will not work, because confetti is not defined. It will work is if confetti is `global variable`. Check this https://www.javatpoint.com/javascript-scope

Reply:
Thank you Claudion3 for this comment. I read about but it I could not find how to make it work. What should I do?

Reply:
Thank you so much for comments, especially this one. i got stuck with this, now it works and I learn about global variable. sooo thanks 👍